packages/block-editor/src/store/reducer.js

```diff
@@ -350,15 +350,18 @@ const withBlockCache = ( reducer ) => ( state = {}, action ) => {
  */
 function withPersistentBlockChange( reducer ) {
 	let lastAction;
+	let markNextChangeAsNotPersistent = false;
 
 	return ( state, action ) => {
 		let nextState = reducer( state, action );
 
-		const isExplicitPersistentChange = action.type === 'MARK_LAST_CHANGE_AS_PERSISTENT';
+		const isExplicitPersistentChange = action.type === 'MARK_LAST_CHANGE_AS_PERSISTENT' || markNextChangeAsNotPersistent;
```

Comment:
This flag seems a little bit confusing because it is true if the change is explicitly persistent or explicitly not persistent, right? The name seems to indicate the flag is true only when the change is explicit persistent, but not when the change is explicitly not persistent.
I guess we could keep the flag as it was before and do them or when we need something explicitly persistent or explicitly not persistent? Or name the flag something like isExplicitPersistentOrNotChange ? (not a very good name but is more detailed).

Reply:
>The name seems to indicate the flag is true only when the change is explicit persistent, but not when the change is explicitly not persistent.

It's an "explicit persistent change". The change could be to `true` or `false`.